tests/unit/test_charm_workload_configuration.py

```diff
@@ -0,0 +1,413 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from io import StringIO
+from pathlib import Path
+from unittest.mock import Mock, call, patch
+
+from charm import AUSFOperatorCharm
+from ops import testing
+
+from lib.charms.tls_certificates_interface.v3.tls_certificates import ProviderCertificate
+
+TEST_POD_IP = b"1.2.3.4"
+
+
+class TestCharmWorkloadConfiguration(unittest.TestCase):
+    def setUp(self):
+        self.namespace = "whatever"
+        self.harness = testing.Harness(AUSFOperatorCharm)
+        self.harness.set_model_name(name=self.namespace)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.set_leader(is_leader=True)
+        self.harness.begin()
+
+    @patch("charm.check_output")
+    @patch("charm.generate_private_key")
+    @patch("ops.model.Container.push")
+    def test_given_charm_workload_is_ready_to_configure_and_private_key_is_not_stored_when_update_status_then_private_key_is_generated_and_stored_in_the_container(  # noqa: E501
+        self, patched_push, patched_generate_private_key, patched_check_output
+    ):
+        test_private_key = b"whatever private key"
+        test_csr = b"whatever csr"
+        self.harness.set_can_connect(container="ausf", val=True)
+        self._create_charm_relations_and_relation_data()
+        self.harness.add_storage(storage_name="certs", attach=True)
+        self.harness.add_storage(storage_name="config", attach=True)
+        patched_check_output.return_value = TEST_POD_IP
+        patched_generate_private_key.return_value = test_private_key
+        root = self.harness.get_filesystem_root("ausf")
+        (root / "support/TLS/ausf.csr").write_text(test_csr.decode())
+
+        self.harness.charm.on.update_status.emit()
+        self.harness.evaluate_status()
+
+        patched_generate_private_key.assert_called_once()
+        patched_push.assert_called_once_with(
+            path="/support/TLS/ausf.key", source=test_private_key.decode()
+        )
```

Comment:
Similar comment as above. We shouldn't need to use patch_puss.assert_xyz

Reply:
Fixed.